File: flutter/library/lib/src/tsp_divider.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

/// Horizontal hairline divider.
class TspDivider extends StatelessWidget {
  const TspDivider({
    super.key,
    this.text = '',
    this.disabled = false,
    this.variant = 'default',
    this.theme = StarPlanetTheme.sky,
  });

  /// Accessibility label only (not rendered as visible text).
  final String text;
  final bool disabled;
  final String variant;
  final StarPlanetTheme theme;

  @override
  Widget build(BuildContext context) {
    return Semantics(
      label: text.isEmpty ? null : text,
      child: Opacity(
        opacity: disabled ? 0.4 : 1,
        child: Container(
          margin: const EdgeInsets.symmetric(vertical: 12),
          height: 1,
          color: variant == 'strong' ? theme.textTertiary : theme.borderDefault,
        ),
      ),
    );
  }
}

File: ios-swiftui/library/Sources/PlanetComponents/TspDivider.swift
import SwiftUI

/// Horizontal hairline divider.
public struct TspDivider: View {
    public var text: String
    public var disabled: Bool
    public var variant: String
    public var theme: StarPlanetTheme

    public init(
        text: String = "",
        disabled: Bool = false,
        variant: String = "default",
        theme: StarPlanetTheme = .sky
    ) {
        self.text = text
        self.disabled = disabled
        self.variant = variant
        self.theme = theme
    }

    public var body: some View {
        Rectangle()
            .fill(variant == "strong" ? theme.textTertiary : theme.borderDefault)
            .frame(height: 1)
            .padding(.vertical, 12)
            .opacity(disabled ? 0.4 : 1)
            .accessibilityLabel(text.isEmpty ? "Divider" : text)
            .accessibilityAddTraits(.isStaticText)
    }
}
